editor.planx.uk/src/@planx/components/ContactInput/Public.tsx

```diff
@@ -38,14 +29,26 @@ export default function ContactInputComponent(props: Props): FCReturn {
       email: "",
     },
     onSubmit: (values) => {
-      // map internal props to expected granular passport keys before submitting
-      values["name.first"] = values.firstName;
-      values["name.last"] = values.lastName;
-      values["company.name"] = values.organisation;
-      values["phone.primary"] = values.phone;
+      // map values to the existing/expected passport structure to minimize conditional handling later in Send schemas, etc
+      const newPassportData: any = {};
+      newPassportData[`${props.fn}.title`] = values.title;
+      newPassportData[`${props.fn}.name.first`] = values.firstName;
+      newPassportData[`${props.fn}.name.last`] = values.lastName;
+      newPassportData[`${props.fn}.company.name`] = values.organisation;
+      newPassportData[`${props.fn}.phone.primary`] = values.phone;
+      newPassportData[`${props.fn}.email`] = values.email;
+
+      const passportData = {
+        [`_contact.${props.fn}`]: { [`${props.fn}`]: values },
```

Comment:
I think I'm missing something here sorry - what's the value / purpose of retaining both structures of contact info?

Reply:
retaining the non-`.` passport variables (eg `firstName`) allows us to pre-fill the formik input fields when going "back" & "changing" !

Reply:
this is part of the stickyness of the original goal to be able to use a name with a `.` in it for the name/value of the input itself - but in lieu of Formik/Yup not particularly liking that - we'll maintain `_contact...` as an internal lookup that is read by `initialValues`, `getPreviouslySubmittedData` etc

Reply:
Perfect - knew there was a reason and there was something I wasn't seeing. Thanks for the explanation ✅